src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,925 @@
+    {
+        std::vector<double> velocity;
+
+
+        // Create the parmeterization to contain the desired number of segments
+        CREATE_STATIC_ARC_LENGTH_PARAMETRIZATION(arc_length_parameterization,
+                                                 TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS);
+
+        float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+        shared_polynomial_getArcLengthParametrizationOrder3(
+            path_parameters.path, path_parameters.t_start, path_parameters.t_end,
+            arc_length_parameterization);
+
+        const float arc_segment_length =
+            shared_polynomial2d_getTotalArcLength(arc_length_parameterization) /
+            path_parameters.num_segments;
+
+
+        app_trajectory_planner_getMaxAllowableSpeedProfile(
+            max_allowable_speed_profile, path_parameters.path,
+            path_parameters.num_segments, arc_length_parameterization, arc_segment_length,
+            path_parameters.max_allowable_acceleration);
+
+        float velocity_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+        velocity_profile[0] = path_parameters.initial_speed;
+        app_trajectory_planner_generateForwardsContinuousVelocityProfile(
+            path_parameters.num_segments, velocity_profile, max_allowable_speed_profile,
+            arc_segment_length, path_parameters.max_allowable_acceleration,
+            path_parameters.max_allowable_speed);
+
+        // Check that it was physically possible for the robot to reach the final velocity
+        // requested
+        if (velocity_profile[path_parameters.num_segments - 1] >=
+            path_parameters.final_speed)
+        {
+            velocity_profile[path_parameters.num_segments - 1] =
+                path_parameters.final_speed;
+        }
+
+        app_trajectory_planner_generateBackwardsContinuousVelocityProfile(
+            path_parameters.num_segments, velocity_profile, arc_segment_length,
+            path_parameters.max_allowable_acceleration);
+
+        velocity_profile[0] = path_parameters.initial_speed;
+
+        for (unsigned int i = 0; i < path_parameters.num_segments; i++)
+        {
+            velocity.push_back(velocity_profile[i]);
+        }
+
+        return velocity;
+    }
+
+    static std::vector<double> getAccelerationsFromSpeed(std::vector<double> speeds,
+                                                         Trajectory_t* trajectory)
+    {
+        std::vector<double> acceleration;
+
+
+        for (unsigned int i = 0; i < trajectory->num_elements - 1; i++)
+        {
+            const double dv = speeds[i + 1] - speeds[i];
+            const double dt = trajectory->trajectory_elements[i + 1].time -
+                              trajectory->trajectory_elements[i].time;
+            acceleration.push_back(dv / dt);
+        }
+
+        return acceleration;
+    }
+};
+
+TEST_F(TrajectoryPlannerTest, test_forward_continuity_curvature_limited)
```

Comment:
It's not immediately obvious that this test is testing the case of "limiting curvature". I'm assuming this is indirectly represented by the max allowable speed being higher than the speed profile. Can we rename this test something like `...speed_profile_limited`?

Reply:
Renamed to "...path_curvature_max_speed_limited"